pkgs/applications/science/electronics/vivado/default.nix

```
@@ -0,0 +1,150 @@
+    message = ''
+      This nix expression requires that ${name} is already part of the store.
+      Login to Xilinx, download from
+      https://www.xilinx.com/support/download.html,
+      rename the file to ${name}, and add it to the nix store with
+      "nix-prefetch-url file:///path/to/${name}".
+    '';
+    sha256 = "06pb4wjz76wlwhhzky9vkyi4aq6775k63c2kw3j9prqdipxqzf9j";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildInputs = [
+    procps
+    ncurses5
+  ];
+
+  postPatch = ''
+    patchShebangs xsetup
+
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+             tps/lnx64/jre/bin/java
+
+    sed -i -- 's|/bin/rm|rm|g' xsetup
+  '';
+
+  dontBuild = true;
+
+  installPhase = ''
+    cat <<EOF > install_config.txt
+    Edition=Vivado HL WebPACK
+    Destination=$out/opt
+    Modules=Software Development Kit (SDK):1,DocNav:0,Kintex UltraScale:0,Zynq-7000:1,System Generator for DSP:0,Artix-7:1,Kintex-7:0
+    InstallOptions=Acquire or Manage a License Key:0,Enable WebTalk for SDK to send usage statistics to Xilinx:0
+    CreateProgramGroupShortcuts=0
+    ProgramGroupFolder=Xilinx Design Tools
+    CreateShortcutsForAllUsers=0
+    CreateDesktopShortcuts=0
+    CreateFileAssociation=0
+    EOF
+
+    mkdir -p $out/opt
+
+    # The installer will be killed as soon as it says that post install tasks have failed.
+    # This is required because it tries to run the unpatched scripts to check if the installation
+    # has succeeded. However, these scripts will fail because they have not been patched yet,
+    # and the installer will proceed to delete the installation if not killed.
+    (./xsetup --agree 3rdPartyEULA,WebTalkTerms,XilinxEULA --batch Install --config install_config.txt || true) | while read line
+    do
+        [[ "''${line}" == *"Execution of Pre/Post Installation Tasks Failed"* ]] && echo "killing installer!" && ((pkill -9 -f "tps/lnx64/jre/bin/java") || true)
+        echo ''${line}
+    done
+  '';
+
+  libPath = lib.makeLibraryPath [
+    stdenv.cc.cc
+    ncurses5
+    zlib
+    libuuid
+    libSM
+    libICE
+    libX11
+    libXrender
+    libxcb
+    libXext
+    libXtst
+    libXi
+    glib
+    freetype
+    gtk2
+  ];
+
+  preFixup = ''
+    # Patch installed files
+    patchShebangs $out/opt/Vivado/$version/bin
+    patchShebangs $out/opt/SDK/$version/bin
+
+    # Hack around lack of libtinfo in NixOS
+    ln -s $ncurses/lib/libncursesw.so.6 $out/opt/Vivado/$version/lib/lnx64.o/libtinfo.so.5
+    ln -s $ncurses/lib/libncursesw.so.6 $out/opt/SDK/$version/lib/lnx64.o/libtinfo.so.5
+
+    # Patch ELFs
+    for f in $out/opt/Vivado/$version/bin/unwrapped/lnx64.o/*
+    do
+        patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $f || true
+    done
+    for f in $out/opt/SDK/$version/bin/unwrapped/lnx64.o/*
+    do
+        patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $f || true
+    done
```

```suggestion
    # Patch ELFs
    for f in $out/opt/Vivado/$version/bin/unwrapped/lnx64.o/*; do
      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $f
    done
    for f in $out/opt/SDK/$version/bin/unwrapped/lnx64.o/*; do
      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $f
    done
```
the build should fail if patchelf fails